service/node/searcher/service.go

```diff
@@ -0,0 +1,68 @@
+package searcher
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/tenantcluster"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const clusterNamespace = "default"
+
+// Config represents the configuration used to create a new service object.
+type Config struct {
+	Logger        micrologger.Logger
+	TenantCluster tenantcluster.Interface
+}
+
+// Service is an object representing the health searcher service.
+type Service struct {
+	logger        micrologger.Logger
+	tenantCluster tenantcluster.Interface
+}
+
+// New creates a new configured service object.
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.TenantCluster == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.TenantCluster must not be empty", config)
+	}
+
+	s := &Service{
+		logger:        config.Logger,
+		tenantCluster: config.TenantCluster,
+	}
+
+	return s, nil
+}
+
+// Search searches for the cluster information.
+// It try to find cluster information in CR and fallback to storage service when nothing is found.
```

Comment:
This comment seems wrong on multiple layers.

Reply:
Oops. Some of these comments have gotten out of sync during copying and pasting. Fixing now.

Reply:
I think I've fixed all of the comments. Going to continue to improve them as I work on other parts.